crates/rolldown/src/runtime/index.js

```diff
@@ -97,11 +97,14 @@ export var __toESM = (mod, isNodeMode, target) => (
     )
 );
 
-// Converts the module from ESM to CommonJS. This clones the input module
-// object with the addition of a non-enumerable "__esModule" property set
-// to "true", which overwrites any existing export named "__esModule".
+// Converts the module from ESM to CommonJS. If the module has a "module.exports"
+// export (Node.js require(ESM) feature), use that directly. Otherwise, clone the input module
+// object with the addition of a non-enumerable "__esModule" property set to "true",
+// which overwrites any existing export named "__esModule".
 export var __toCommonJS = mod =>
-  __copyProps(__defProp({}, '__esModule', { value: true }), mod);
+  __hasOwnProp.call(mod, 'module.exports')
+    ? mod['module.exports']
+    : __copyProps(__defProp({}, '__esModule', { value: true }), mod);
```

Comment:
@hyf0 @IWANABETHATGUY Does this look good to you?
For reference, Node's implementation is here: https://github.com/nodejs/node/pull/54563/files#diff-5b5902273122e094ff474fda358605ffa45a4a58b51cd0bf4c1acb93779df142R1430-R1440